lib/Controllers/list.js

```diff
@@ -95,8 +95,12 @@ List.prototype.fetch = function(req, res, context) {
     return result;
   }, {});
 
-  if (Object.keys(extraSearchCriteria).length)
-    criteria = _.assign(extraSearchCriteria, criteria);
+  if (Object.keys(extraSearchCriteria).length) {
+    if (Object.keys(criteria).length)
+      criteria = Sequelize.and(criteria, Sequelize.or.apply(null, [extraSearchCriteria]));
+    else
+      criteria = Sequelize.or.apply(null, [extraSearchCriteria]);
```

Comment:
Yesterday got busy, but I should have time to update the PR today. I'll keep you posted.